Playback stops on short segment

### What version of Hls.js are you using?
v0.13.2

### What browser and OS are you using?
MacOS 10.15.2, Chrome 81.0.4044.138 (Official Build) (64-bit)

### Test stream:
<!-- If possible, please provide a test stream or page -->
<!-- You can paste your stream into the demo and provide the permalink here -->
https://hls-js.netlify.app/demo/?src=https%3A%2F%2Fstorage.googleapis.com%2Fmihir-vod-mrss%2Fkaios-all-pos-dai-conditioned-ads%2Fmaster.m3u8&demoConfig=eyJlbmFibGVTdHJlYW1pbmciOnRydWUsImF1dG9SZWNvdmVyRXJyb3IiOnRydWUsImR1bXBmTVA0IjpmYWxzZSwibGV2ZWxDYXBwaW5nIjotMSwibGltaXRNZXRyaWNzIjotMX0=

### Checklist
<!-- Replace [ ] with [x] to check off the list -->
- [x] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [x] The issue occurs in the stable client on https://hls-js.netlify.com/demo and not just on my page
<!-- The stable client is built from the latest release -->
- [x] The issue occurs in the latest client on https://hls-js-latest.netlify.com/demo and not just on my page
<!-- The latest client is built from the head of the master branch -->
- [x] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [x] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce
1. Please provide clear steps to reproduce your problem
Play provided stream.

2. If the bug is intermittent, give a rough frequency.
Happens every time.

### Expected behavior
*What you expected to happen*
Successful playback till end of video. E.g., Safari plays back the stream just fine.

### Actual behavior
*What actually happened*
Stream freezes at the 60s mark. This coincides to a short 33ms segment. Looking at the network trace you can notice hls.js does not even attempt to download this short segment.

Note that removing this 33ms segment from the manifest allows the playback to complete successfully, e.g.: https://storage.googleapis.com/mihir-vod-mrss/kaios-new-all-pos-dai-cond-no-small-seg/master.m3u8

### Console output
```
[log] > destroy
logger.js:40 [log] > detachMedia
logger.js:40 [log] > media source detaching
main.js:216 Using Hls.js config: {debug: true, enableWorker: true, liveBackBufferLength: 900}
logger.js:40 [log] > loadSource:https://storage.googleapis.com/mihir-vod-mrss/kaios-all-pos-dai-conditioned-ads/master.m3u8
logger.js:40 [log] > trigger BUFFER_RESET
logger.js:40 [log] > set autoLevelCapping:-1
logger.js:40 [log] > attachMedia
logger.js:40 [log] > media source opened
logger.js:40 [log] > manifest loaded,1 level(s) found, first bitrate:270010
logger.js:40 [log] > 1 bufferCodec event(s) expected
logger.js:40 [log] > startLoad(-1)
logger.js:40 [log] > switching to level 0
logger.js:40 [log] > main stream-controller: STOPPED->IDLE
logger.js:40 [log] > audio tracks updated
logger.js:40 [log] > subtitle tracks updated
logger.js:40 [log] > level 0 loaded [1,20],duration:115.0145
logger.js:40 [log] > Loading 1 of [1 ,20],level 0, currentTime:0.000,bufferEnd:0.000
logger.js:40 [log] > demuxing in webworker
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 1 of [1 ,20],level 0
logger.js:40 [log] > Parsing 1 of [1 ,20],level 0, cc 0
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > main:discontinuity detected
logger.js:40 [log] > main:switch detected
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > manifest codec:undefined,ADTS data:type:2,sampleingIndex:7[22050Hz],channelConfig:2
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > parsed codec:mp4a.40.5,rate:22050,nb channel:2
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > audio sampling rate : 22050
logger.js:40 [log] > InitPTS for cc: 0 found from video track: 0
logger.js:40 [log] > creating sourceBuffer(audio/mp4;codecs=mp4a.40.5)
logger.js:40 [log] > creating sourceBuffer(video/mp4;codecs=avc1.4d400d)
logger.js:40 [log] > main track:audio,container:audio/mp4,codecs[level/parsed]=[undefined/mp4a.40.5]
logger.js:40 [log] > main track:video,container:video/mp4,codecs[level/parsed]=[undefined/avc1.4d400d]
logger.js:40 [log] > Parsed audio,PTS:[0.000,5.016],DTS:[0.000/5.016],nb:108,dropped:0
logger.js:40 [log] > Parsed video,PTS:[0.033,5.005],DTS:[0.000/5.005],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,5.016]
logger.js:40 [log] > latency/loading/parsing/append/kbps:42/93/19/3/8265
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 2 of [1 ,20],level 0, currentTime:5.005,bufferEnd:5.016
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Updating Media Source duration to 115.014
logger.js:40 [log] > Loaded 2 of [1 ,20],level 0
logger.js:40 [log] > Parsing 2 of [1 ,20],level 0, cc 0
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
favicon.ico:1 GET https://hls-js.netlify.app/favicon.ico 404
logger.js:40 [log] > Parsed audio,PTS:[5.016,10.031],DTS:[5.016/10.031],nb:108,dropped:0
logger.js:40 [log] > Parsed video,PTS:[5.038,10.043],DTS:[5.005/10.010],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,10.031]
logger.js:40 [log] > latency/loading/parsing/append/kbps:41/29/10/1/15611
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 3 of [1 ,20],level 0, currentTime:0.010,bufferEnd:10.031
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 3 of [1 ,20],level 0
logger.js:40 [log] > Parsing 3 of [1 ,20],level 0, cc 0
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > Parsed audio,PTS:[10.031,15.000],DTS:[10.031/15.000],nb:107,dropped:0
logger.js:40 [log] > Parsed video,PTS:[10.043,15.048],DTS:[10.010/15.015],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,15.000]
logger.js:40 [log] > latency/loading/parsing/append/kbps:29/51/7/1/13998
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 5 of [1 ,20],level 0, currentTime:0.072,bufferEnd:15.000
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 5 of [1 ,20],level 0
logger.js:40 [log] > Parsing 5 of [1 ,20],level 0, cc 1
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > main:discontinuity detected
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > manifest codec:undefined,ADTS data:type:2,sampleingIndex:7[22050Hz],channelConfig:2
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > parsed codec:mp4a.40.5,rate:22050,nb channel:2
127ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > audio sampling rate : 22050
logger.js:40 [log] > InitPTS for cc: 1 found from video track: -457323
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [warn] > Injecting 2 audio frames @ 15000 ms due to 15000 ms gap.
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562
remuxAudio @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4291
remux @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:3816
append @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:2136
pushDecrypted @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4809
push @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4738
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:132
logger.js:40 [log] > main track:audio,container:audio/mp4,codecs[level/parsed]=[undefined/mp4a.40.5]
logger.js:40 [log] > main track:video,container:video/mp4,codecs[level/parsed]=[undefined/avc1.4d400d]
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [warn] > PTS < DTS detected in video samples, shifting DTS by -33 ms to overcome this issue
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562
remuxVideo @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4009
remux @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:3831
append @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:2136
pushDecrypted @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4809
push @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4738
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:132
logger.js:40 [log] > Parsed audio,PTS:[15.000,25.078],DTS:[15.000/25.078],nb:217,dropped:0
logger.js:40 [log] > Parsed video,PTS:[15.081,25.025],DTS:[15.048/25.025],nb:299,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,25.058]
logger.js:40 [log] > latency/loading/parsing/append/kbps:29/100/39/2/15364
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 6 of [1 ,20],level 0, currentTime:0.244,bufferEnd:25.058
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 6 of [1 ,20],level 0
logger.js:40 [log] > Parsing 6 of [1 ,20],level 0, cc 1
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
127ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
logger.js:40 [log] > Parsed audio,PTS:[25.078,35.062],DTS:[25.078/35.062],nb:215,dropped:0
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [warn] > PTS < DTS detected in video samples, shifting DTS by -33 ms to overcome this issue
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562
remuxVideo @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4009
remux @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:3831
append @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:2136
pushDecrypted @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4809
push @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4738
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:132
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > Video: PTS/DTS adjusted: 25058/25025, delta: 0 ms
logger.js:40 [log] > Parsed video,PTS:[25.058,35.068],DTS:[25.025/35.001],nb:299,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,35.062]
logger.js:40 [log] > latency/loading/parsing/append/kbps:432/78/25/1/4870
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 7 of [1 ,20],level 0, currentTime:0.782,bufferEnd:35.062
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 7 of [1 ,20],level 0
logger.js:40 [log] > Parsing 7 of [1 ,20],level 0, cc 1
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
127ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AAC: align PTS for overlapping frames by -46
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [warn] > PTS < DTS detected in video samples, shifting DTS by -33 ms to overcome this issue
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562
remuxVideo @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4009
remux @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:3831
append @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:2136
pushDecrypted @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4809
push @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4738
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:132
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > Video: PTS/DTS adjusted: 35001/35001, delta: 0 ms
logger.js:40 [log] > Parsed audio,PTS:[35.062,45.000],DTS:[35.062/45.000],nb:214,dropped:0
logger.js:40 [log] > Parsed video,PTS:[35.001,44.978],DTS:[35.001/44.978],nb:299,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,45.000]
logger.js:40 [log] > latency/loading/parsing/append/kbps:37/143/29/2/12414
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 8 of [1 ,20],level 0, currentTime:0.996,bufferEnd:45.000
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 8 of [1 ,20],level 0
logger.js:40 [log] > Parsing 8 of [1 ,20],level 0, cc 1
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [warn] > PTS < DTS detected in video samples, shifting DTS by -33 ms to overcome this issue
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562
remuxVideo @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4009
remux @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:3831
append @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:2136
pushDecrypted @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4809
push @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:4738
(anonymous) @ 7ff933ae-078e-40bf-8b18-6fbf7bf9636e:132
logger.js:40 [log] > Parsed audio,PTS:[45.000,45.140],DTS:[45.000/45.140],nb:3,dropped:0
logger.js:40 [log] > Parsed video,PTS:[45.045,45.111],DTS:[44.978/45.078],nb:3,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,45.111]
logger.js:40 [log] > latency/loading/parsing/append/kbps:70/0/2/1/370
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 9 of [1 ,20],level 0, currentTime:1.072,bufferEnd:45.111
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 9 of [1 ,20],level 0
logger.js:40 [log] > Parsing 9 of [1 ,20],level 0, cc 2
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > main:discontinuity detected
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > manifest codec:undefined,ADTS data:type:2,sampleingIndex:7[22050Hz],channelConfig:2
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > parsed codec:mp4a.40.5,rate:22050,nb channel:2
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > audio sampling rate : 22050
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > AVC: 62 ms hole between fragments detected,filling it
7ff933ae-078e-40bf-8b18-6fbf7bf9636e:562 [log] > Video: PTS/DTS adjusted: 45111/45078, delta: 62 ms
logger.js:40 [log] > InitPTS for cc: 2 found from video track: -4062563.265306106
logger.js:40 [log] > main track:audio,container:audio/mp4,codecs[level/parsed]=[undefined/mp4a.40.5]
logger.js:40 [log] > main track:video,container:video/mp4,codecs[level/parsed]=[undefined/avc1.4d400d]
logger.js:40 [log] > Parsed audio,PTS:[45.140,50.155],DTS:[45.140/50.155],nb:108,dropped:0
logger.js:40 [log] > Parsed video,PTS:[45.111,50.145],DTS:[45.078/50.145],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,50.155]
logger.js:40 [log] > latency/loading/parsing/append/kbps:76/34/6/1/11061
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 10 of [1 ,20],level 0, currentTime:1.193,bufferEnd:50.155
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 10 of [1 ,20],level 0
logger.js:40 [log] > Parsing 10 of [1 ,20],level 0, cc 2
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > Parsed audio,PTS:[50.155,55.171],DTS:[50.155/55.171],nb:108,dropped:0
logger.js:40 [log] > Parsed video,PTS:[50.178,55.183],DTS:[50.145/55.150],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,55.171]
logger.js:40 [log] > latency/loading/parsing/append/kbps:58/39/3/1/12428
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [log] > Loading 11 of [1 ,20],level 0, currentTime:1.296,bufferEnd:55.171
logger.js:40 [log] > main stream-controller: IDLE->FRAG_LOADING
logger.js:40 [log] > Loaded 11 of [1 ,20],level 0
logger.js:40 [log] > Parsing 11 of [1 ,20],level 0, cc 2
logger.js:40 [log] > main stream-controller: FRAG_LOADING->PARSING
logger.js:40 [log] > Parsed audio,PTS:[55.171,60.140],DTS:[55.171/60.140],nb:107,dropped:0
logger.js:40 [log] > Parsed video,PTS:[55.183,60.188],DTS:[55.150/60.155],nb:150,dropped:0
logger.js:40 [log] > main stream-controller: PARSING->PARSED
logger.js:40 [log] > main buffered : [0.033,60.140]
logger.js:40 [log] > latency/loading/parsing/append/kbps:106/60/4/1/7177
logger.js:40 [log] > main stream-controller: PARSED->IDLE
logger.js:40 [warn] > Playback stalling at @60.088556 due to low buffer
(anonymous) @ logger.js:40
_reportStall @ gap-controller.js:160
poll @ gap-controller.js:106
_checkBuffer @ stream-controller.js:1271
doTick @ stream-controller.js:129
tick @ task-loop.ts:111
setInterval (async)
setInterval @ task-loop.ts:73
startLoad @ stream-controller.js:57
(anonymous) @ hls.ts:301
startLoad @ hls.ts:300
onManifestParsed @ stream-controller.js:751
onEventGeneric @ event-handler.ts:82
onEvent @ event-handler.ts:69
emit @ index.js:203
trigger @ observer.ts:13
onManifestLoaded @ level-controller.js:150
onEventGeneric @ event-handler.ts:82
onEvent @ event-handler.ts:69
emit @ index.js:203
trigger @ observer.ts:13
_handleMasterPlaylist @ playlist-loader.ts:327
loadsuccess @ playlist-loader.ts:267
readystatechange @ xhr-loader.js:120
XMLHttpRequest.send (async)
(anonymous) @ VM10:1
loadInternal @ xhr-loader.js:82
load @ xhr-loader.js:40
load @ playlist-loader.ts:235
onManifestLoading @ playlist-loader.ts:132
onEventGeneric @ event-handler.ts:82
onEvent @ event-handler.ts:69
emit @ index.js:203
trigger @ observer.ts:13
loadSource @ hls.ts:288
loadSelectedStream @ main.js:222
(anonymous) @ main.js:67
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
main.js:539 Error event: {type: "mediaError", details: "bufferStalledError", fatal: false, buffer: 0.05108000000000601}
```

--- a/src/controller/fragment-finders.ts
+++ b/src/controller/fragment-finders.ts
@@ -46,12 +46,23 @@ export function findFragmentByPDT (fragments: Array<Fragment>, PDTValue: number 
  * @returns {*} foundFrag - The best matching fragment
  */
 export function findFragmentByPTS (fragPrevious: Fragment, fragments: Array<Fragment>, bufferEnd: number = 0, maxFragLookUpTolerance: number = 0): Fragment | null {
-  const fragNext = fragPrevious ? fragments[fragPrevious.sn as number - (fragments[0].sn as number) + 1] : null;
+  let fragNext: Fragment | null = null;
+  if (fragPrevious) {
+    fragNext = fragments[fragPrevious.sn as number - (fragments[0].sn as number) + 1];
+  } else if (bufferEnd === 0 && fragments[0].start === 0) {
+    fragNext = fragments[0];
+  }
   // Prefer the next fragment if it's within tolerance
-  if (fragNext && !fragmentWithinToleranceTest(bufferEnd, maxFragLookUpTolerance, fragNext)) {
+  if (fragNext && fragmentWithinToleranceTest(bufferEnd, maxFragLookUpTolerance, fragNext) === 0) {
     return fragNext;
   }
-  return BinarySearch.search(fragments, fragmentWithinToleranceTest.bind(null, bufferEnd, maxFragLookUpTolerance));
+  // We might be seeking past the tolerance so find the best match
+  const foundFragment = BinarySearch.search(fragments, fragmentWithinToleranceTest.bind(null, bufferEnd, maxFragLookUpTolerance));
+  if (foundFragment) {
+    return foundFragment;
+  }
+  // If no match was found return the next fragment after fragPrevious, or null
+  return fragNext;
 }
 
 /**

--- a/tests/test-streams.js
+++ b/tests/test-streams.js
@@ -101,7 +101,7 @@ module.exports = {
   },
   */
   mp3Audio: {
-    'url': 'https://player.webvideocore.net/CL1olYogIrDWvwqiIKK7eLBkzvO18gwo9ERMzsyXzwt_t-ya8ygf2kQBZww38JJT/8i4vvznv8408.m3u8',
+    'url': 'https://playertest.longtailvideo.com/adaptive/vod-with-mp3/manifest.m3u8',
     'description': 'MP3 VOD demo',
     'live': false,
     'abr': false,

--- a/tests/unit/controller/fragment-finders.js
+++ b/tests/unit/controller/fragment-finders.js
@@ -39,6 +39,30 @@ describe('Fragment finders', function () {
       expect(binarySearchSpy).to.have.not.been.called;
     });
 
+    it('chooses the fragment with the next SN if its contiguous with the end of previous fragment', function () {
+      // See https://github.com/video-dev/hls.js/issues/2776
+      const bufferEnd = 60.139636;
+      const fragments = [{
+        deltaPTS: 0.012346258503441732,
+        cc: 2,
+        duration: 5.017346258503444,
+        start: 55.21705215419478,
+        sn: 11,
+        level: 0
+      },
+      {
+        deltaPTS: 0,
+        cc: 2,
+        duration: 0.033,
+        start: 60.234398412698226,
+        sn: 12,
+        level: 0
+      }];
+      const fragPrevious = fragments[0];
+      const actual = findFragmentByPTS(fragPrevious, fragments, bufferEnd, tolerance);
+      expect(actual).to.equal(fragments[1], `expected sn ${fragments[1].sn}, but got sn ${actual ? actual.sn : null}`);
+    });
+
     it('uses BinarySearch to find a fragment if the subsequent one is not within tolerance', function () {
       const fragments = [mockFragments[0], mockFragments[(mockFragments.length - 1)]];
       findFragmentByPTS(fragments[0], fragments, bufferEnd, tolerance);

